### PR TITLE
Update Shopcart API

### PR DESCRIPTION
**Pull request by LillisLuo**
Complete the route logic of Update Shopcart API in **routes.py**
Wrote two test cases in **test.models.py**
Wrote five test cases in **test.routes.py**
截屏2023-07-04 20 54 33
截屏2023-07-04 20 57 10
截屏2023-07-04 20 57 30

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -205,3 +206,12 @@ def deserialize(self, data: dict) -> None:
             raise DataValidationError(
                 f"Invalid {type(self).__name__}: failed to deserialize request body\nError message: {error}"
             ) from error
+        
+    @classmethod
+    def get_dp_id(cls, shopcart_id, item_id):
```
Comment (wl2484) [inline service/models.py:211]: would you like to reuse get_by_id()? or add corresponding test case for this? lol

File: tests/test_models.py [hunk 2]
```diff
@@ -68,6 +68,65 @@ def test_read_an_shopcart(self):
...
+        item.update()
+
+        shopcart = Shopcart.get_by_id(shopcart.id)
+        self.assertEqual(len(shopcart.items), 1)
+        item_name =  shopcart.items[0].name
+        item_price =  shopcart.items[0].price
+        item_quantity =  shopcart.items[0].quantity
+        self.assertEqual(item.name, item_name)
+        self.assertEqual(item.price, item_price)
+        self.assertEqual(item.quantity, item_quantity)
```
Comment (wl2484) [inline tests/test_models.py:130]: You may refer to this [commit](https://github.com/CSCI-GA-2820-SU23-001/shopcarts/pull/47/commits/e0f0fc7ab2dafd3086940cd23705b1c17cd9b047) to add a test case specifically for `shopcart=None`. With this, line 133 in `routes.py` would no longer be counted as a line without test!

**Review by wl2484 — COMMENTED** [review]
cool!

File: service/models.py (hunk 1 above)
Comment (LillisLuo) [reply service/models.py:211]: Yes, I forgot to delete some part that didn't use. I will check the code again before merge

Comment (wl2484) [reply service/models.py:211]: no problem! thank you~~

**Comment by wl2484** [comment]
btw, there's conflicts with the master branch 🥲 please help to resolve it, thanks a lot!

**Comment by LillisLuo** [comment]
> btw, there's conflicts with the master branch 🥲 please help to resolve it, thanks a lot!

Yes, I will figure this out
